polkadot/node/overseer/src/lib.rs

```diff
@@ -743,6 +743,26 @@ where
 	async fn run_inner(mut self) -> SubsystemResult<()> {
 		let metrics = self.metrics.clone();
 		spawn_metronome_metrics(&mut self, metrics)?;
+		let spawner = self.spawner.clone();
+		let mut to_orchestra_rx =
+			self.to_orchestra_rx.take().expect("Orchestrate always sets this to some; qed");
+
+		let orchestra_rx = Box::pin(async move {
```

Comment:
worth adding a comment why (preventing deadlock) and extracting this out to be a separate PR

Reply:
It isn't appear to be solving the problem, but seems like a sane thing to do anyway to make ctx.spawn() safe to use generally.